### ISSUE TITLE
Movie review sentiment analysis

### ISSUE THREAD

**Issue by bdzyubak** [bdzyubak/torch-control#14]
Fine-tune Distilbert for movie sentiment review on the following dataset: 
https://www.kaggle.com/competitions/sentiment-analysis-on-movie-reviews/data

The IMDB dataset is also interesting for sentiment review. Potentially, implement as a separate experiment and then cross validate training on one or both. Implement the other common networks and compare performance. For those that come with out-of-the-box sentiment analysis, evaluate performance without fine-tuning.

**Comment by bdzyubak** [comment]
DistilBERT was fine tuned to achieve 0.8 training accuracy and 0.68 validation accuracy. The peak validation loss accuracy was achieved in one epoch after which point validation performance deteriorated severely as the model overfit the training data and lost its generalized pre-trained weights. Due to checkpointing, the best model at the first epoch was saved, so we don't need to worry about the later epochs. 

![training_metrics_version_7](https://github.com/bdzyubak/torch-control/assets/37943739/8ffa866b-fa37-4985-be21-44334841f6b1)

Next steps: 
1) The IMDB dataset is also interesting for sentiment review. Potentially, implement as a separate experiment and then cross validate training on one or both. 
2) Implement the other common networks and compare performance. For those that come with out-of-the-box sentiment analysis, evaluate performance without fine-tuning. 
3) Compare fine-tuning with frozen layers and only the sentiment head being allowed to train.
